### PR TITLE
Update find-refs to find references to nint/nuint

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/42988

Properly understands if nint/nuint are true aliases to IntPtr/UIntPtr or not depending on underlying framework.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/PredefinedTypeExtensions.cs [hunk 1]
```diff
@@ -9,46 +9,28 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
...
+                PredefinedType.UInt32 => SpecialType.System_UInt32,
+                PredefinedType.Int64 => SpecialType.System_Int64,
+                PredefinedType.UInt64 => SpecialType.System_UInt64,
+                PredefinedType.Decimal => SpecialType.System_Decimal,
+                PredefinedType.Single => SpecialType.System_Single,
+                PredefinedType.Double => SpecialType.System_Double,
+                PredefinedType.String => SpecialType.System_String,
+                PredefinedType.DateTime => SpecialType.System_DateTime,
+                PredefinedType.IntPtr => SpecialType.System_IntPtr,
+                PredefinedType.UIntPtr => SpecialType.System_UIntPtr,
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/PredefinedTypeExtensions.cs:32]: last two lines are added.

File: src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb [hunk 2]
```diff
@@ -945,5 +945,262 @@ namespace X
...
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+using System;
+class Test
+{
+    void M()
+    {
+        $${|Reference:nint|} n1 = 1;
```
Comment (Cosifne) [inline src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb:961]: Why the behavior changes when `CommonReferencesNet7=true`?

Comment (Cosifne) [reply src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb:961]: Oh I just realize this is the design decision of .net 7

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb:961]: :)

Comment (Cosifne) [reply src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb:961]: How does this actually work when the runtime is different? (i.e. I am surprised with how this work, are the symbols we get from the compiler side different?)

**Review by Cosifne — APPROVED** [review]
Some minor questions. Other than that lgtm

File: src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb:961]: So the compiler produces different symbols.  They differ with this property:

https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs#L82

On .net7 and above, both of these will return true for IsNativeIntegerType.  Prior to .net 7, only nint/nuint would, and IntPtr/UIntPtr would return false.

**Pull request merged**